service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/any/flexibleresource/FlexResourceCreationParameters.java

```diff
@@ -0,0 +1,28 @@
+package bio.terra.workspace.service.resource.controlled.cloud.any.flexibleresource;
+
+import bio.terra.workspace.generated.model.ApiControlledFlexibleResourceCreationParameters;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import javax.annotation.Nullable;
+
+public record FlexResourceCreationParameters(
+    String typeNamespace, String type, @Nullable byte[] data) {
+
+  @JsonCreator
+  public FlexResourceCreationParameters(
```

Comment:
Do we need json serilization of this class? in another word, can we not have this constructor? (kinda defeat the purpose of making it into a record)

we can add @JsonSerialize if we must. https://dev.to/brunooliveira/practical-java-16-using-jackson-to-serialize-records-4og4

Reply:
I don't think so. I can remove this constructor. We'll see later if we need the JSON serialization (probably not).

Reply:
The structure will be serialized when used as a Stairway parameter. Part of my goal here is to use internally defined classes for everything we serialize in Stairway. It can be hard to correlate an API change to a deserialization error in a flight.

Jackson is very good at serdes of Java records. You shouldn't need a special serializer